openelevationservice/server/grpc/grpc_server.py

```diff
@@ -0,0 +1,144 @@
+            geom_queried = querybuilder.polygon_elevation(geom, 'polygon', 'srtm')
+            geom_shaped = wkt.loads(geom_queried)
+            
+            result = []
+            for point in list(geom_shaped.coords):
+                result.append(defs.LatLonElevation(
+                    lon=point[0],
+                    lat=point[1],
+                    elevation=int(point[2])
+                ))
+
+            return defs.AreaPointsResponse(points=result)
+    
+    def _create_proto_geo_polygon(self, coordinates):
+        return defs.Area(boundaries=[
+            defs.LineString(points=[
+                defs.LatLon(
+                    lon=point[0],
+                    lat=point[1],
+                ) for point in bondary
+            ]) for bondary in coordinates            
+        ])
+
+    def AreaRangesElevation(self, request, context):
+        with self.app.app_context():
+            geom = convert.polygon_to_geometry(self._format_area_request(request))
+            collection_queried, range_queried = querybuilder.polygon_coloring_elevation(geom, 'srtm')
+            
+            result = []
+            for feature in collection_queried['features']:
+                heightBase = int(feature['properties']['heightBase'])
+                if feature['geometry']['type'] == 'Polygon':
+                    result.append(defs.UnitedArea(
+                        baseElevation=heightBase,
+                        area=self._create_proto_geo_polygon(feature['geometry']['coordinates']),
+                    ))
+                else:
+                    for polygon in feature['geometry']['coordinates']:
+                        result.append(defs.UnitedArea(
+                            baseElevation=heightBase,
+                            area=self._create_proto_geo_polygon(polygon),
+                        ))
+            
+            return defs.AreaRangesResponse(
+                unions=result,
+                minElevation=range_queried[0],
+                maxElevation=range_queried[1]
+            )
+
+
+def grpc_serve(app, port_url):
+    server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
+    openelevation_pb2_grpc.add_OpenElevationServicer_to_server(
+        OpenElevationServicer(app), server)
+    
+    SERVICE_NAMES = (
+        defs.DESCRIPTOR.services_by_name['OpenElevation'].full_name,
+        reflection.SERVICE_NAME,
+    )
+
+    reflection.enable_server_reflection(SERVICE_NAMES, server)
+    
+    # TODO: use correct credentials if needed
+    # if '-s' in sys.argv[1:]:
+    #     privkey = open('./test_ssl/test_key.pem', 'rb').read()
+    #     certchain = open('./test_ssl/test_cert.pem', 'rb').read()
+    #     server.add_secure_port(
+    #         port_url,
+    #         grpc.ssl_server_credentials(
+    #             ((privkey, certchain), ),
+    #             # root_certificates=None,
+    #             # require_client_auth=False
+    #         )
+    #     )
+    # else:
```

Comment:
Perhaps we should create an issue based on this TODO so that we don't forget about it, and we'll see later whether to use SSL with our certificates.

Reply:
#15